target/linux/ipq807x/files/arch/arm64/boot/dts/qcom/ipq8072-ax880.dts

```diff
@@ -0,0 +1,390 @@
+// SPDX-License-Identifier: MIT
+/* Copyright (c) 2023, Ruslan Isaev <legale.legale@gmail.com> */
+
+/dts-v1/;
+
+#include "ipq8074.dtsi"
+#include "ipq8074-hk-cpu.dtsi"
+#include "ipq8074-ess.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Yuncore AX880";
+	compatible = "yuncore,ax880", "qcom,ipq8074";
+
+	aliases {
+		serial0 = &blsp1_uart5;
+		serial1 = &blsp1_uart3;
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+		/* Aliases as required by u-boot to patch MAC addresses */
+		ethernet0 = &dp5_syn;
+		ethernet1 = &dp6_syn;
+		label-mac-device = &dp5_syn;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+        bootargs-append = " root=/dev/ubiblock0_1";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&button_pins>;
+		pinctrl-names = "default";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 57 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_system: system {
+			label = "system";
+			gpios = <&tlmm 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan2g {
+			label = "wlan2g";
+			gpio = <&tlmm 2 GPIO_ACTIVE_HIGH>;
+		};
+
+		wlan5g {
+			label = "wlan5g";
+			gpio = <&tlmm 9 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&tlmm {
+	mdio_pins: mdio-pins {
+		mdc {
+			pins = "gpio68";
+			function = "mdc";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+
+		mdio {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+
+	button_pins: button_pins {
+		reset_button {
+			pins = "gpio57";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+};
+
+&blsp1_spi1 {
+	status = "okay";
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		reg = <0>;
+		compatible = "jedec,spi-nor";
+		spi-max-frequency = <50000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
```

Comment:
Are the partitions defined in SMEM incorrect?

Reply:
These are correct for sure.

Reply:
Just to make, can you please post the output of smem in u-boot to see if the SMEM table is correct or broken?

Reply:
> Just to make, can you please post the output of smem in u-boot to see if the SMEM table is correct or broken?

```
IPQ807x# smem
flash_type:             0x6
flash_index:            0x0
flash_chip_select:      0x0
flash_block_size:       0x10000
flash_density:          0x1000000
partition table offset  0x0
No.: Name             Attributes            Start             Size
  0: 0:SBL1           0x0000ffff              0x0          0x50000
  1: 0:MIBIB          0x001040ff          0x50000          0x10000
  2: 0:BOOTCONFIG     0x001040ff          0x60000          0x20000
  3: 0:BOOTCONFIG1    0x001040ff          0x80000          0x20000
  4: 0:QSEE           0x0000ffff         0x220000         0x180000
  5: 0:QSEE_1         0x0000ffff          0xa0000         0x180000
  6: 0:DEVCFG         0x0000ffff         0x3b0000          0x10000
  7: 0:DEVCFG_1       0x0000ffff         0x3a0000          0x10000
  8: 0:APDP           0x0000ffff         0x3c0000          0x10000
  9: 0:APDP_1         0x0000ffff         0x3d0000          0x10000
 10: 0:RPM            0x0000ffff         0x420000          0x40000
 11: 0:RPM_1          0x0000ffff         0x3e0000          0x40000
 12: 0:CDT            0x0000ffff         0x470000          0x10000
 13: 0:CDT_1          0x0000ffff         0x460000          0x10000
 14: 0:APPSBLENV      0x0000ffff         0x480000          0x10000
 15: 0:APPSBL         0x0000ffff         0x530000          0xa0000
 16: 0:APPSBL_1       0x0000ffff         0x490000          0xa0000
 17: 0:ART            0x0000ffff         0x5d0000          0x40000
 18: 0:ETHPHYFW       0x0000ffff         0x610000          0x80000
 19: rootfs           0x0100ffff        0x3c00000        0x3400000
 20: 0:WIFIFW         0x0100ffff        0x7000000         0x800000
 21: rootfs_1         0x0100ffff              0x0        0x3400000
 22: 0:WIFIFW_1       0x0100ffff        0x3400000         0x800000
IPQ807x#
```

Reply:
If i remove these partiions in dts there is no chance to work with them from initramfs, so i think it is better to leave them.

Reply:
Okay I see, using smem would lead to overlapping partitions.
But any particular reason why you are missing lowercase and uppercase for the partition labels?

In my opinion you should use lowercase for all labels.

Reply:
I agree, please lowercase them so userspace scripts dont have to be changed just because of upper case.

I get the issue, it will try to use the same table for both NOR and NAND as there is no flash type parsing.